Lib/multiprocessing/popen_forkserver.py

```diff
@@ -49,7 +49,11 @@ def _launch(self, process_obj):
             set_spawning_popen(None)
 
         self.sentinel, w = forkserver.connect_to_new_process(self._fds)
+        # Keep a duplicate of the data pipe's write end as a sentinel of the
+        # parent process used by the child process.
+        self._parent_w = os.dup(w)
         self.finalizer = util.Finalize(self, os.close, (self.sentinel,))
+        self.finalizer = util.Finalize(self, os.close, (self._parent_w,))
```

Comment:
`_parent_w` is not the best name, but I did not find convincing, short alternatives. I thought `_child_sentinel` was somewhat unclear. Any ideas anyone?

Reply:
However, you're also clobbering the previous finalizer here.